### DISCUSSION TITLE
DTO vs Entity

### DISCUSSION THREAD

**Discussion by xorinzor** [abpframework/abp#14782] [Q&A] [answered]
Hey All,

I'm new to working with ABP, having mostly worked with frameworks for PHP like Laravel.

I'm trying to understand the relationship between DTO's and Entities, as they seemingly contain the same properties, yet I cannot fetch / insert Entities from my Application Service as apparently I need to use DTO's for that.

This confuses me, as to me, it seems like I'm effectively defining the same information, twice, to reach the same outcome.
This is probably the incorrect way to think about it, but I'm having difficulty understanding how these objects interact. The `BookStore` example did not really help clear this up for me either.

Could someone explain this to me?

**Comment by realLiangshiwei** [comment]
You can see the EBook: https://abp.io/books/implementing-domain-driven-design

**Reply by xorinzor** [reply]
This doesn't tell me anything that the documentation on the website doesn't already has written.
I know DTO's are supposed to transfer data between the Application and Presentation layers, what I don't know (and isn't explained) is why you effectively define the same code twice by adding the same properties of the Entity to the DTO.

Generally whenever you have to define code twice, you're doing something wrong.

**Comment by EngincanV** [comment, marked as answer]
Hi @xorinzor, please check the [The Need for DTOs](https://docs.abp.io/en/abp/latest/Data-Transfer-Objects#the-need-for-dtos) section in the [Data Transfer Objects](https://docs.abp.io/en/abp/latest/Data-Transfer-Objects) documentation, why we need to use DTOs is explained in detail.

**Reply by xorinzor** [reply]
Thanks, that clarified it for me :)
